crates/hir-def/src/nameres/path_resolution.rs

```diff
@@ -488,8 +513,21 @@ impl DefMap {
                             ),
                         })
                     });
+                    // FIXME: Need to filter visibility here and below? Not sure.
                     match res {
-                        Some(res) => res,
+                        Some(res) => {
+                            if segments.next().is_some() {
+                                // Enum variants are in value namespace, segments left => no resolution.
+                                return ResolvePathResult::empty(ReachedFixedPoint::No);
+                            }
+                            return ResolvePathResult::new_enum_variant(
+                                res,
+                                ReachedFixedPoint::Yes,
+                                None,
+                                false,
+                                i - 1,
```

Comment:
That is regarding my previous comment, this is an incorrect assumption, the prefix of a path does not need to be an enum, it can be a module if the variant was imported directly and then referred to via a module path prefix.